### ISSUE TITLE
executing uno watch finished the execution

### ISSUE THREAD

**Issue by jjolmo** [zposten/unorepo#5]
Hi. 
I'm using uno so it launches yarn build whenever I modify a project in my monorepo. But when I launch `uno watch` in the monorepo root it lists correctly all the packages that will be watched, but it does nothing. It ends the execution of the script.

Is there a way to debug what is actually happening?

Thanks!

**Comment by 0livare** [comment]
Hi @jjolmo,

I appreciate you bringing up the issue, this will be interesting.  This is the first problem with unorepo that I haven't been able to walk over to someone's desk to address.

I have a few questions:
  - Is the project you're working on open source so that I can look at it directly?
  - Are you on a unix or windows operating system?
  - Does the script exit immediately after listing the files, or does is exit once you make a change to a project file?
  - Just to confirm, it doesn't show any error, correct?
  - You're on version 0.0.25, correct?

**Comment by hirikarate** [comment]
Dear @zposten,
I have the same problem. My monorepo structure is:
```
|__ lerna.json
|__ package.json
|__ packages
   |__ apps
      |__ service-A
      |__ service-B
      |__ service-c
   |__ libs
     |__ common
     |__ models
     |__ utilities
```

My root `package.json` looks partially like this. Note that I must escape double quotes instead of single quotes because of PowerShell.
```json
{
   "scripts": {
      "watch": "uno watch --ext ts --execute \"tsc --pretty\"",
   }
}
```

My lerna.json:
```json
{
	"lerna": "3.16.4",
	"command": {
		"publish": {
			"allowBranch": ["master"]
		}
	},
	"packages": [
		"packages/apps/*",
		"packages/libs/*"
	],
	"npmClient": "yarn",
	"version": "independent"
}

```
The output is _literally_ like this. The script exits immediately after listing the files. Wierd thing is that only the last "libs" package is listed:
```powershell
my-project> yarn watch
yarn run v1.17.3
$ uno watch --ext ts --execute "tsc --pretty"
[uno]  Watching ["*.ts"] from the following packages:
[uno]   utilities
Done in 3.10s.
```

My environment:
- Unorepo: 0.0.25
- OS: Windows 10 64-bit

My project is not opensource, if you have difficulties in re-produce it, I will create a Github repo.

**Comment by 0livare** [comment]
@hirikarate Thank you for the additional information!  With those details I believe that I was able to reproduce your exact problem and [fix](https://github.com/zposten/unorepo/commit/97308b8efe3c73159c7ac50e64807d183e510785) the issue.  

The problem was your use of a "libs" directory.  I had originally chosen to ignore several typical build directory names to avoid triggering unnecessary cycles.  However, given your feedback I've decided to remove that mechanic.

The issue should be fixed if you install `v0.0.26` of unorepo

```bash
yarn add --dev unorepo@latest
```

**Comment by hirikarate** [comment]
Thank you for quick responses, I created a [(mono)repository to demostrate this problem](https://github.com/hirikarate/unorepo-watching-bugs).

Problem:
1. Not all packages are watched
2. Script is not executed when watched packages are changed.

Here are my opinions after debugging into unorepo source code:

* **Prob 1:** Some of my company packages have "private=true" in package.json, so unorepo must execute `lerna ls --all` with arguments `--all` to get full list of packages.
* **Prob 2:** Arg `--ext` is converted to glob `{path}/*.ts`, while I believe it should be `{path}/**/*.ts` to match arbitrary levels of children directories.

**Comment by 0livare** [comment]
@hirikarate Thank you for the example monorepo and for digging into the problems you're having.

In response to your comments, I have added two new flags, `--ignore` and `--include-private`.  

`--ignore` addresses an issue I created with my original attempt to fix this bug.  Originally as I said I had removed code that ignored a certain set of build directories.  But I quickly realized that doing so means that unorepo can get into an infinite loop of changes when a build happens because it detects the changes to those built files (e.g. `.d.ts` files).  So my fix for this is to still provide a default list of common build directories (with "lib" removed to prevent issues with monorepos like yours), but to allow customization of that list with the `--ignore` flag.

`--include-private` does exactly what you expect, when it's passed private packages will also be watched.

So for your use case, with version `v0.0.28` you should just have to include the `--include-private` flag and everything should hopefully work as you expect.  I tested using your example monorepo and everything seemed to work!

Rather than prematurely closing the issue again, I'm going to reopen it.  Please let me know if this has indeed fixed your issue, and then we can close this.

**Comment by hirikarate** [comment]
The private packages are correctly included, however changes are not detected. I digged around your code and found this piece of gold in `chokidar`'s [documentation](https://github.com/paulmillr/chokidar#api)

> Note: globs must not contain windows separators (\), because that's how they work by the standard — you'll need to replace them with forward slashes (/).

Therefore I tried adding some code to replace back slashes `\` with forward slashes `/` in `watch.js` and now it works on my Windows 10 machine.

Please see the attached image for what I changed to make it work.

![image](https://user-images.githubusercontent.com/5905553/71431623-b9939c80-2705-11ea-82e9-eb059a1873f0.png)

**Comment by 0livare** [comment]
@hirikarate Nice find!  

I made that fix and published it to version `0.0.29-beta.0`.  Can you try installing it with `yarn add --dev unorepo@beta`, and let me know if it works?  Everything seemed pleasantly unchanged on my unix machine.

If it works as expected, I'll publish the real 0.0.29!
